modules/engage-theodul-plugin-tab-downloads/src/main/resources/static/main.js

```diff
@@ -0,0 +1,216 @@
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+/*jslint browser: true, nomen: true*/
+/*global define*/
+/*global require*/
+define(['jquery', 'underscore', 'backbone', 'engage/core'], function ($, _, Backbone, Engage) {
+  'use strict';
+
+  var insertIntoDOM = false;
+  var PLUGIN_NAME = 'Downloads';
+  var PLUGIN_TYPE = 'engage_tab';
+  var PLUGIN_VERSION = '1.0';
+  var PLUGIN_TEMPLATE_DESKTOP = 'templates/desktop.html';
+  var PLUGIN_STYLES_DESKTOP = [
+    'styles/desktop.css'
+  ];
+
+  var isDesktopMode = Engage.model.get('mode') === 'desktop';
+
+  // The following code block contains the logic for the download tab in the theodul player.
+  // The download tab should only be displayed if it's explicitly allowed in the 'config.yml' file and
+  // when the user has a role that matches a specific regex. The regex is also configured in the 'config.yml'.
+  const video_download_allowed = Engage.model.get('meInfo').get('allow_video_download');
+  if (video_download_allowed) {
+    const userRoles = Engage.model.get('meInfo').get('roles');
+    const roleRegex = new RegExp(Engage.model.get('meInfo').get('allowed_roles_for_video_download'));
+    const roleMatches = userRoles.some(function (role) {
+      return roleRegex.exec(role);
+    });
+    if (roleMatches) {
+      insertIntoDOM = true;
+    }
+  }
+
+  var events = {
+    plugin_load_done: new Engage.Event('Core:plugin_load_done', '', 'handler'),
+    mediaPackageModelError: new Engage.Event('MhConnection:mediaPackageModelError', '', 'handler')
+  };
+
+  var plugin = {
+    insertIntoDOM: insertIntoDOM,
+    name: PLUGIN_NAME,
+    type: PLUGIN_TYPE,
+    version: PLUGIN_VERSION,
+    styles: PLUGIN_STYLES_DESKTOP,
+    template: PLUGIN_TEMPLATE_DESKTOP,
+    events: events
+  };
+
+  /* don't change these variables */
+  var viewsModelChange = 'change:views';
+  var mediapackageChange = 'change:mediaPackage';
+  var initCount = 4;
+  var mediapackageError = false;
+  var translations = new Array();
+  var Utils;
+
+
+  function initTranslate(language, funcSuccess, funcError) {
+    var path = Engage.getPluginPath('EngagePluginTabDownloads').replace(/(\.\.\/)/g, '');
+    var jsonstr = window.location.origin + '/engage/theodul/' + path; // this solution is really bad, fix it...
+
+    if (language == 'de') {
+      Engage.log('Tab:Downloads: Chosing german translations');
+      jsonstr += 'language/de.json';
+    } else { // No other languages supported, yet
+      Engage.log('Tab:Downloads: Chosing english translations');
+      jsonstr += 'language/en.json';
+    }
```

Comment:
There can be many more languages. This should probably be something like:

```suggestion
    Engage.log('Tab:Downloads: Chosing ' + language + ' translations');
    jsonstr += 'language/' + language + '.json';
```

Reply:
Or even just:

```js
const url = window.location.origin + '/engage/theodul/' + path + 'language/' + language + '.json';
```

But even easier, you can just use a path without domain for the request:

```js
const url = '/engage/theodul/' + path + 'language/' + language + '.json';
```

and you could also use a template string:

```js
const url = `/engage/theodul/${path}language/${language}.json`;
```

Reply:
I used this version now: `const url = '/engage/theodul/${path}language/${language}.json'`